searchcore/src/vespa/searchcore/proton/persistenceengine/resource_usage_tracker.h

```diff
@@ -0,0 +1,33 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <vespa/searchcore/proton/server/i_disk_mem_usage_listener.h>
+#include <vespa/persistence/spi/resource_usage.h>
+#include <mutex>
+#include <memory>
+#include <vector>
+
+namespace storage::spi { class IResourceUsageListener; }
+
+namespace proton {
+
+class DiskMemUsageState;
+class IDiskMemUsageNotifier;
+
+class ResourceUsageTracker : public IDiskMemUsageListener
```

Comment:
Please add class description.

Reply:
This class is currently not unit tested, but I guess that will be part of upcoming PRs.